vendor/tooltipscan.lua

```diff
@@ -37,6 +37,50 @@ local function importTooltipTextToTable(tooltip, text, bag, slot)
     return tooltipTable
 end
 
+
+
+local function importTooltipColorToTable(tooltip, side, bag, slot)
+    -- We assume the tooltip is the GameTooltip
+    local tooltipText = "GameTooltipText"..side
+
+    -- If we don't have the gametooltip, use the scanning tooltip and use the bag item.
+    if not tooltip then
+        tooltip = scanningtip
+        tooltip:ClearLines()
+        tooltipText = "VendorScanningTipText"..side
+        if bag and slot then
+            tooltip:SetBagItem(bag, slot)
+        else
+            error("Invalid arguments to Tooltip Import")
+        end
+    end
+
+    -- Import the tooltip into a table
+    local tooltipTable = {}
```

Comment:
Instead of an import, this should be a boolean function that returns true if the tooltip has red text. Other text colors are not interesting, so a one time scan as part of the item to read this makes sense to me.